### PR TITLE
#2638 iframe csp

### PR DESCRIPTION
**Pull request by AlexeyTuboltsev**
this PR adds csp headers/url sanitization to twitter and slideshare iframes

### CODE REVIEW

File: packages/web/app/external/sanitize-url/sanitize-url.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+var invalidPrototcolRegex = /^(%20|\s)*(javascript|data)/im;
+var ctrlCharactersRegex = /[^\x20-\x7EÀ-ž]/gim;
+var urlSchemeRegex = /^([^:]+):/gm;
+var relativeFirstCharacters = [".", "/"];
+
+function isRelativeUrlWithoutProtocol(url) {
+  return relativeFirstCharacters.indexOf(url[0]) > -1;
+}
+
+function sanitizeUrl(url) {
```
Comment (desfero) [inline packages/web/app/external/sanitize-url/sanitize-url.js:14]: maybe some unit tests to make sure it's working properly always?

File: packages/web/app/external/pitch-deck-iframe.html [hunk 2]
```diff
@@ -3,6 +3,10 @@
   <head>
     <meta charset="utf-8" />
     <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
+    <meta
+      http-equiv="Content-Security-Policy"
+      content="default-src 'self'; frame-src *; script-src 'self' 'sha256-80vXT2q9DtdrNkWOqKM45NjhN0z2WCwccwwrAQbJw2E='; style-src 'unsafe-inline'; img-src 'self';"
```
Comment (desfero) [inline packages/web/app/external/pitch-deck-iframe.html:8]: What kind of js file `sha256-80vXT2q9DtdrNkWOqKM45NjhN0z2WCwccwwrAQbJw2E=` represents?

File: packages/web/app/components/shared/TwitterTimeline.tsx [hunk 3]
```diff
@@ -16,8 +16,11 @@ export class TwitterTimelineEmbed extends React.Component<IProps> {
         width="100%"
         height="100%"
         frameBorder={0}
-        allowFullScreen
+        allowFullScreen={false}
```
Comment (desfero) [inline packages/web/app/components/shared/TwitterTimeline.tsx:19]: maybe we can completely omit the attribute if it's false?

File: packages/web/app/external/pitch-deck-iframe.html (hunk 2 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/external/pitch-deck-iframe.html:8]: I think it represents  scripts inside <script></script> tags. 
see here: 
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src

that means effectively that even if everything else fails only this script will be run

File: packages/web/app/components/shared/TwitterTimeline.tsx (hunk 3 above)
Comment (AlexeyTuboltsev) [reply packages/web/app/components/shared/TwitterTimeline.tsx:19]: absolutely

File: packages/web/app/external/sanitize-url/sanitize-url.spec.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+  it("disregards capitalization for JavaScript urls", () => {
+    expect(sanitizeUrl("jAvasCrIPT:alert(document.domain)")).to.eq("");
+  });
+
+  it("ignores ctrl characters in javascript urls", () => {
+    expect(sanitizeUrl(decodeURIComponent("JaVaScRiP%0at:alert(document.domain)"))).to.eq("");
+  });
+
+  it("replaces javascript urls with  when javascript url begins with %20", () => {
```
[the remark this answers is not in the document]
Comment (AlexeyTuboltsev) [reply packages/web/app/external/sanitize-url/sanitize-url.spec.js:19]: ups

**Pull request merged**